apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -278,8 +278,25 @@ type BareMetalHostSpec struct {
 	// the power status and hardware inventory inspection. If the
 	// Image field is filled in, this field is ignored.
 	ExternallyProvisioned bool `json:"externallyProvisioned,omitempty"`
+
+	// When set to Disabled, automated cleaning will be avoided
+	// during provisioning and deprovisioning.
+	// +optional
+	// +kubebuilder:default:=Full
+	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode AutomatedCleaningMode `json:"automatedCleaningMode,omitempty"`
 }
 
+// AutomatedCleaningMode is the interface to enable/disable automated cleaning
+// +kubebuilder:validation:Enum:=Full;Disabled
+type AutomatedCleaningMode string
+
+// Allowed automated cleaning modes
+const (
+	CleaningModeDisabled AutomatedCleaningMode = "Disabled"
+	CleaningModeFull     AutomatedCleaningMode = "Full"
```

Comment:
Maybe not call it "Full" in case we ever support actually removing all data, not just the partitioning table? We in ironic call it "metadata cleaning".

Reply:
I can definitely rename it if desired. Any thoughts on this @dhellmann ?

Reply:
Future-proofing the name would be good.

If we're only cleaning metadata, but on all devices, should we reflect that? `MetadataAllDevices` seems a bit long but at least it's descriptive. We could then have `MetadataSelectedDevices` at some point and variations for `AllDataAllDevices` and `AllDataSelectedDevices` as well.

@dtantsur Is the switch to control whether we delete metadata or all data exposed in the Ironic API, or just a config setting?

Are we ever likely to support different modes per device?

Reply:
> Future-proofing the name would be good.
> 
> If we're only cleaning metadata, but on all devices, should we reflect that? `MetadataAllDevices` seems a bit long but at least it's descriptive. We could then have `MetadataSelectedDevices` at some point and variations for `AllDataAllDevices` and `AllDataSelectedDevices` as well.
> 
Maybe for now just `Metadata` ? Because, if we are going to extend it in the future, we will anyways have to touch the API. As such, maybe we could leave it to be done in future and for this proposal just stick to the `Metadata` ?

Reply:
OK, I can go along with `Metadata` applying to all devices, for now. And in the future we can have `MetadataSelectedDevices` or something like that.

Reply:
@dhellmann Updated the PR and the [proposal](https://github.com/metal3-io/metal3-docs/pull/166) as well.